### ISSUE TITLE
Segmentation fault outputting to file more than 2048 records

### ISSUE THREAD

**Issue by minaguib** [duckdb/duckdb#6204]
### What happens?

duckdb CLI crashes when output to file exceeds 2048 rows with:
```
Segmentation fault: 11
```

Under lldb: 
```
Process 10060 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x1000400)
(lldb) thread backtrace
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x1000400)
  * frame #0: 0x000000010060a268 duckdb`duckdb::Vector::GetValueInternal(duckdb::Vector const&, unsigned long long) + 188
    frame #1: 0x000000010060b238 duckdb`duckdb::Vector::GetValue(duckdb::Vector const&, unsigned long long) + 24
    frame #2: 0x00000001006c61c4 duckdb`duckdb::BoxRenderer::GetRenderValue(duckdb::ColumnDataRowCollection&, unsigned long long, unsigned long long) + 48
    frame #3: 0x00000001006c7900 duckdb`duckdb::BoxRenderer::RenderValues(std::__1::list<duckdb::ColumnDataCollection, std::__1::allocator<duckdb::ColumnDataCollection> > const&, std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> > const&, std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> > const&, std::__1::vector<duckdb::LogicalType, std::__1::allocator<duckdb::LogicalType> > const&, std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 720
    frame #4: 0x00000001006c42cc duckdb`duckdb::BoxRenderer::Render(duckdb::ClientContext&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, duckdb::ColumnDataCollection const&, std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 980
    frame #5: 0x00000001006c3ce0 duckdb`duckdb::BoxRenderer::ToString(duckdb::ClientContext&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, duckdb::ColumnDataCollection const&) + 84
    frame #6: 0x000000010001d004 duckdb`duckdb_shell_sqlite3_print_duckbox + 1072
    frame #7: 0x000000010001778c duckdb`exec_prepared_stmt + 1356
    frame #8: 0x000000010000cd14 duckdb`shell_exec + 488
    frame #9: 0x00000001000185c4 duckdb`runOneSqlLine + 156
    frame #10: 0x000000010000d9d0 duckdb`process_input + 1004
    frame #11: 0x0000000100005ba8 duckdb`main + 3944
    frame #12: 0x000000018d7c3e50 dyld`start + 2544
```

### To Reproduce

```
$ duckdb
v0.6.1 919cad22e8
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D .output foo.txt
D select * from range(2048); -- OK
D select * from range(2049); -- Will crash
Segmentation fault: 11
```

### OS:

MacOS Ventura 13.2 (22D49)

### DuckDB Version:

v0.6.1 919cad22e8

### DuckDB Client:

duckdb_cli

### Full Name:

Mina Naguib

### Affiliation:

Hivestack

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! The box renderer used indeed does not seem to deal well with rendering a large amount of values. When rendering to a file the output is not truncated anymore. I have pushed a fix in #6205. 

As a work-around you could use a different rendering mode, e.g.:

```
.mode csv
select * from range(2049); 
```

**Comment by minaguib** [comment]
@Mytherin Amazing. Thank you (for the fix, and the amazing product in general).
